osu.Game.Rulesets.Taiko/Difficulty/TaikoPerformanceCalculator.cs

```diff
@@ -87,35 +91,98 @@ private double computeDifficultyValue(ScoreInfo score, TaikoDifficultyAttributes
                 difficultyValue *= 1.025;
 
             if (score.Mods.Any(m => m is ModHardRock))
-                difficultyValue *= 1.050;
+                difficultyValue *= 1.10;
 
             if (score.Mods.Any(m => m is ModFlashlight<TaikoHitObject>))
                 difficultyValue *= 1.050 * lengthBonus;
 
-            return difficultyValue * Math.Pow(accuracy, 2.0);
+            if (estimatedUr == null)
+                return 0;
+
+            return difficultyValue * Math.Pow(SpecialFunctions.Erf(400 / (Math.Sqrt(2) * estimatedUr.Value)), 2.0);
         }
 
         private double computeAccuracyValue(ScoreInfo score, TaikoDifficultyAttributes attributes, bool isConvert)
         {
-            if (attributes.GreatHitWindow <= 0)
+            if (attributes.GreatHitWindow <= 0 || estimatedUr == null)
                 return 0;
 
-            double accuracyValue = Math.Pow(60.0 / attributes.GreatHitWindow, 1.1) * Math.Pow(accuracy, 8.0) * Math.Pow(attributes.StarRating, 0.4) * 27.0;
+            double accuracyValue = Math.Pow(60 / estimatedUr.Value, 1.1) * Math.Pow(attributes.StarRating, 0.4) * 100.0;
 
             double lengthBonus = Math.Min(1.15, Math.Pow(totalHits / 1500.0, 0.3));
-            accuracyValue *= lengthBonus;
 
             // Slight HDFL Bonus for accuracy. A clamp is used to prevent against negative values.
             if (score.Mods.Any(m => m is ModFlashlight<TaikoHitObject>) && score.Mods.Any(m => m is ModHidden) && !isConvert)
-                accuracyValue *= Math.Max(1.0, 1.1 * lengthBonus);
+                accuracyValue *= Math.Max(1.0, 1.05 * lengthBonus);
 
             return accuracyValue;
         }
 
+        /// <summary>
+        /// Estimates the player's tap deviation based on the OD, number of objects, and number of 300s, 100s, and misses,
+        /// assuming the player's mean hit error is 0. The estimation is consistent in that two SS scores on the same map with the same settings
+        /// will always return the same deviation. See: https://www.desmos.com/calculator/qlr946netu
+        /// </summary>
+        private double? computeEstimatedUr(ScoreInfo score, TaikoDifficultyAttributes attributes)
+        {
+            if (totalSuccessfulHits == 0 || attributes.GreatHitWindow == 0)
+                return null;
+
+            // Create a new track to properly calculate the hit window of 100s.
+            var track = new TrackVirtual(10000);
+            score.Mods.OfType<IApplicableToTrack>().ForEach(m => m.ApplyToTrack(track));
+            double clockRate = track.Rate;
+
+            double overallDifficulty = (50 - attributes.GreatHitWindow * clockRate) / 3;
+            double h300 = attributes.GreatHitWindow;
+            double h100 = overallDifficulty <= 5 ? (120 - 8 * overallDifficulty) / clockRate : (80 - 6 * (overallDifficulty - 5)) / clockRate;
```

Comment:
Assuming this is not a custom OD calculation, maybe we can just use `HitWindow` here, or also store the hit window for 100s (`MehHitWindow`?) in `TaikoDifficultyAttribute`?

Reply:
this would work, i was avoiding it as it's a redundant difficulty attribute but if that's the direction you want to go i'm fine with it

Reply:
@Lawtrohux What do you think? If we don't want to add another attribute we can probably do something like `new HitWindow(od).WindowFor(HitResult.Meh)`, assuming `ScoreInfo.BeatmapInfo.BeatmapDifficulty.OverallDifficulty` is available

On the other hand, I would say different hit windows aren't completely redundant. From my testing while writing rhythm, they scale differently with OD (i.e. OD with half the 300 hit window won't have half the 50 hit window). Which also means that DT would make things different too. No idea if that extra information would be useful as a display though

Reply:
Hmm would you prefer adding meh hit window in attributes for this pr, or leave it as is for now and address it in a future pr?

One counterargument though: From my understandings, it is (along with `GreatHitWindow`) redundant in the sense that all scores will have these as attributes, which will be identical given the same map and mod combination. Intuitively this should be in `DifficultyAttribute` instead, though I don't know if `PerformanceCalculator` have access to that.

Reply:
On a second thought, ignore the second paragraph, I seem to have mixed up the concept, as it was always passed in as `DifficultyAttribute`

Reply:
I'm now passing the ok hit window as a separate difficulty attribute, since that's the direction both of you want to go